src/vivarium_public_health/disease/special_disease.py

```diff
@@ -171,16 +168,20 @@ def setup(self, builder):
         self.disability_weight = builder.value.register_value_producer(
             f"{self.cause.name}.disability_weight",
             source=self.compute_disability_weight,
+            component=self,
             requires_columns=get_lookup_columns(
```

Comment:
Isn't part of this ticket to also get rid of all of these deprecated args and replace w/ `required_resources`?